org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/console/JavaStackTraceHyperlink.java

```diff
@@ -203,6 +234,232 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
+						if (methodSignatureInnerClass.equals(methodSignatureGen)) {
+							return true;
+						}
+					}
+					return false;
+				}
+			}
+		} catch (Exception e) {
+			return false;
+		}
+		return false;
+	}
+
+	/**
+	 * Additional method signature generation for inner classes
+	 *
+	 * @param extractedMethodName
+	 *            Extracted method name from input
+	 * @param methodSignature
+	 *            generated method signature
+	 * @return returns generated <code>String</code> suitable for inner classes
+	 */
+	private String innerClassMethodSignatureGen(String extractedMethodName, String methodSignature) {
+		StringBuilder newSignature = new StringBuilder(extractedMethodName + "("); //$NON-NLS-1$
+		String paramsExtracted = methodSignature.substring(methodSignature.indexOf('(') + 1, methodSignature.indexOf(')'));
+		if (paramsExtracted.indexOf(',') != -1) {
+			String[] parameters = paramsExtracted.split(","); //$NON-NLS-1$
+			for (String param : parameters) {
+				newSignature.append(param.substring(param.indexOf('$') + 1));
+				newSignature.append(","); //$NON-NLS-1$
+			}
+			newSignature.deleteCharAt(newSignature.length() - 1);
+			if (newSignature.charAt(newSignature.length() - 1) != ')') {
+				newSignature.append(')');
+			}
+		}
+		return newSignature.toString();
+	}
+	/**
+	 * Method signature generation for normal classes
+	 *
+	 * @param methodName
+	 *            Extracted method name from input
+	 * @param targetMethod
+	 *            already extracted method signature
+	 * @return returns generated <code>String</code> suitable for normal classes
+	 */
+	private String methodSignatureGenerator(String methodName, String targetMethod) {
+		StringBuilder methodSignatureBuilder = new StringBuilder(methodName);
+		methodSignatureBuilder.append('(');
+		String[] params = targetMethod.split(","); //$NON-NLS-1$
+		for (String block : params) {
+			if (block.contains("...")) { //$NON-NLS-1$ Parameter is var args
+				if (params.length > 1) {
+					methodSignatureBuilder = varArgsParamBuilder(block, methodSignatureBuilder);
+				}
+			} else {
+				if (block.indexOf('.') == -1) {
+					methodSignatureBuilder.append(block.substring(block.lastIndexOf('(') + 1));
+					methodSignatureBuilder.append(',');
+				} else {
+					methodSignatureBuilder.append(block.substring(block.lastIndexOf('.') + 1));
+					methodSignatureBuilder.append(',');
+				}
+			}
+		}
+		methodSignatureBuilder.deleteCharAt(methodSignatureBuilder.length() - 1);
+
+		if (methodSignatureBuilder.charAt(methodSignatureBuilder.length() - 1) != ')') {
+			methodSignatureBuilder.append(')');
+		}
+		return methodSignatureBuilder.toString();
+	}
+	/**
+	 * Variable argument's parameter creation
+	 *
+	 * @param parameter
+	 *            Input parameter
+	 * @param current
+	 *            Initial parameters
+	 * @return returns generated <code>Stringbuilder</code> for parameter
+	 */
+	private StringBuilder varArgsParamBuilder(String parameter, StringBuilder current) {
```

Comment:
Method is not tested: the test case with varargs is missing

Reply:
Added now.